apps/finance/contracts/Finance.sol

```diff
@@ -2,24 +2,22 @@
  * SPDX-License-Identitifer:    GPL-3.0-or-later
  */
 
-pragma solidity 0.4.18;
+pragma solidity 0.4.24;
 
 import "@aragon/os/contracts/apps/AragonApp.sol";
 
-import "@aragon/os/contracts/lib/zeppelin/token/ERC20.sol";
-import "@aragon/os/contracts/lib/zeppelin/math/SafeMath.sol";
-import "@aragon/os/contracts/lib/zeppelin/math/SafeMath64.sol";
+import "@aragon/os/contracts/common/EtherTokenConstant.sol";
+import "@aragon/os/contracts/lib/token/ERC20.sol";
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+import "@aragon/os/contracts/lib/math/SafeMath64.sol";
 
 import "@aragon/apps-vault/contracts/IVaultConnector.sol";
 
-import "@aragon/os/contracts/lib/misc/Migrations.sol";
 
-
-contract Finance is AragonApp {
+contract Finance is AragonApp, EtherTokenConstant {
```

Comment:
`AragonApp` is always `EtherTokenConstant`, so I'll remove it, ok?

Reply:
Mmm, I've generally been prefering to keep the "hidden" classes in the explicit inheritance tree, especially in the case of things like shell contracts solely meant to share constants and the like:

1. It explicitly tells readers that we're using the functionality, rather than guessing where it might come from
1. In case it gets removed from a upstream class, nothing breaks. In this case, it's not even `AragonApp` that's inheriting it, but `VaultRecoverable`.

Both of the above are intuitions from the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter).

But this is definitely a bit of a gray zone. I think I prefer to keep it here, because, applying LoD, `AragonApp` doesn't make it obvious that it includes `EtherTokenConstant`. As another example, if a contract were to be `Petrifiable` but also use `TimeHelpers`, I'd include both even though `Petrifiable` uses `TimeHelpers` under the hood, as it's not immediately obvious that `Petrifiable` is related.

An example in the other direction is something like `AppStorage`. If you're using an `AragonApp`, I wouldn't include `AppStorage` again if you're explicitly using the `apps` state as the two are closely coupled.

Reply:
A little related annoyance is the problem with C3 linearization. Trying to put it back the way it was made Solidity fail to compile after merging #410. Same as here I guess: https://github.com/aragon/aragon-apps/pull/256#discussion_r211939204

Reply:
I guess we could adopt the rule of adding them from “most base-like” to “most derived”, as it's recommended in Solidity documentation: https://solidity.readthedocs.io/en/v0.4.24/contracts.html#multiple-inheritance-and-linearization

Reply:
Yeah... the order will have to be swapped. Admittedly it makes it not very nice :(.

It's hard to really tell what's "most base-like" some times, but with unstructured storage this is less of a big deal than before.